amethyst_input/src/bundle.rs

```diff
@@ -127,40 +118,46 @@ where
 
 /// An error occurred while loading the bindings file.
 #[derive(Debug)]
-pub enum BindingsFileError<AX, AC> {
+pub enum BindingsFileError<AX: 'static, AC: 'static> {
```

Comment:
I imagine it's because otherwise the type can't be upcast.

Reply:
Yes, what @Xaeroxe said. I'm not sure if it's a strict requirement, or just the type system being overzealous. But in order for the type to satisfy `dyn (std::error::Error + 'static)` all generic parameters have to be `'static` as well.

Reply:
Boxed trait objects tend to assume their contents contain no lifetimes that aren't mentioned in the trait definition. Otherwise you could create an unsafe situation wherein a boxed trait object holds references to something that's been dropped. So this is more than a requirement of Rust, it's a technical requirement.